### PR TITLE
Auto code shows msgs & surveys and generate production file

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Hi @as2388  PTAL. Have fixed minor naming issues in pipeline config, upgraded core data module to v0.9.0 and added auto code shows msgs & surveys and generate production file. Pending https://github.com/AfricasVoices/Project-ICRAF/pull/2 https://github.com/AfricasVoices/Project-ICRAF/pull/3 https://github.com/AfricasVoices/Project-ICRAF/pull/4 https://github.com/AfricasVoices/Project-ICRAF/pull/5 https://github.com/AfricasVoices/Project-ICRAF/pull/6 reviews

### CODE REVIEW

**Comment by IsaackMwenda** [comment]
@as2388  Due to major changes/conflicts in the project I closed https://github.com/AfricasVoices/Project-ICRAF/pull/6 and addressed all the issues in that PR here. In addition this branch has the autocode messages, generation of production file . PTAL

File: configurations/pipeline_config.json
Comment (as2388) [inline configurations/pipeline_config.json]: Do you mean new_practices_challenge*s*_raw?

(I don't know, just checking!)

File: pipeline.py
Comment (as2388) [inline pipeline.py:198]: If you're committing commented code you should add a TODO explaining why. 

It's super obvious to me in this case, but it also helps to remind you to uncomment it later.

File: pipeline.py
Comment (as2388) [inline pipeline.py]:
```suggestion
                "'DriveUploadPaths')")
```

File: src/auto_code_show_messages.py
Comment (as2388) [inline src/auto_code_show_messages.py]: I think your indentation is wrong here

File: src/lib/pipeline_configuration.py
Comment (as2388) [inline src/lib/pipeline_configuration.py]: Put this comment on the preceding line and start it with "TODO".

Then it becomes easier to see when scanning the file, and you get editor support

Comment (as2388) [reply src/lib/pipeline_configuration.py]: Might be worth setting the start/end dates to datetime.min/.max to make it more obvious that these dates are temporary

File: src/lib/pipeline_configuration.py
Comment (as2388) [inline src/lib/pipeline_configuration.py]: Why are these called follow-up coding plans? Aren't these the main radio question flows?

(Previous projects that I have worked on have had rqa flows which handle the answers to each week's radio question, demographic profiling flows which run after we receive an rqa answer, and then follow-up surveys, which are a bit like demographics but ask follow-on questions about the radio shows, and are sent at set times after the rqa flows. I think I'm confused as to whether you're using "follow-up" in this usual AVF sense or in some new sense unique to this project)

File: src/lib/channels.py
Comment (as2388) [inline src/lib/channels.py]: Can you submit this file as a separate PR? This is so we can additionally have someone from RDA as a reviewer of the timestamps

File: src/lib/channels.py
Comment (as2388) [inline src/lib/channels.py]: If you do this then you're ignoring messages received at e.g. 2019-04-04T23:59:59.500+03:00.
The last end date is exclusive, so instead you can say 2019-04-04T24:00:00+03:00, which means anything before the end of day on April 4th.

**Review by as2388 — CHANGES_REQUESTED** [review]

File: src/lib/pipeline_configuration.py
Comment (IsaackMwenda) [reply src/lib/pipeline_configuration.py]: Your right, Its the same context as previous projects. Have combined RQA and Follow up coding plans in one list so that I can apply some of the autocode and apply manual code methods on them without having to rewrite the methods twice. Will give the coding plan a more clear name RQA_&_FOLLOW_UP_CODING_PLANS

File: src/lib/channels.py
Comment (IsaackMwenda) [reply src/lib/channels.py]: Great thanks for this.

File: src/lib/channels.py
Comment (IsaackMwenda) [reply src/lib/channels.py]: Okay will do thanks

File: pipeline.py
Comment (IsaackMwenda) [reply pipeline.py:198]: Nice tip thanks!

File: src/lib/pipeline_configuration.py
Comment (IsaackMwenda) [reply src/lib/pipeline_configuration.py]: Nice tips. Updated thanks alot.

File: src/lib/channels.py
Comment (IsaackMwenda) [reply src/lib/channels.py]: Have updated to ("2019-04-02T00:00:00+0300", "2019-04-05T07:00:00+0300"), To cover promos that may be aired early morning before the show.

File: src/auto_code_show_messages.py
Comment (IsaackMwenda) [reply src/auto_code_show_messages.py]: Nice catch thanks. Updated!

File: configurations/pipeline_config.json
Comment (IsaackMwenda) [reply configurations/pipeline_config.json]: Thats right because one may answer more than one challenge. Updated!

File: pipeline.py
Comment (IsaackMwenda) [reply pipeline.py]: Fixed!

File: src/lib/pipeline_configuration.py
Comment (as2388) [reply src/lib/pipeline_configuration.py]: Is this definitely the right place to put the follow up surveys (the other option being in the same list as the demographics)?

I think it depends on where you want to put that data in the individuals analysis csv i.e. do you have a new row per answer like the RQAs, or fill in a new column for every RQA response, like we do with the demographics? It probably also depends on whether you're expecting one response to each follow-up survey or whether people can send repeat replies to the same question. What does RDA need for this project?

Comment (IsaackMwenda) [reply src/lib/pipeline_configuration.py]: Discussed offline.

File: docker-run.sh
Comment (as2388) [inline docker-run.sh:13]: Low priority, but this flag isn't needed anymore now that this is defined in pipeline_config.json. I don't mind you merging with it still included, provided you file an issue against the repo to remove it.

File: pipeline.py
Comment (as2388) [inline pipeline.py]:
```suggestion
                     s01e05_input_path, s01e06_input_path]
```

File: src/apply_manual_codes.py
Comment (as2388) [inline src/apply_manual_codes.py]: This was changed to RQA + FOLLOW_UP in a future PR right?

If not, don't merge this.

File: src/lib/channels.py
Comment (as2388) [reply src/lib/channels.py]: Remove this file from here before merging or, if it breaks the pipeline, delete the timestamps.

(This is because I want the timestamps to be introduced in PR https://github.com/AfricasVoices/Project-ICRAF/pull/11)

**Review by as2388 — APPROVED** [review]

File: src/apply_manual_codes.py
Comment (IsaackMwenda) [reply src/apply_manual_codes.py]: Erroneously committed here. Fixed in future pr

File: src/lib/channels.py
Comment (IsaackMwenda) [reply src/lib/channels.py]: File deleted

File: docker-run.sh
Comment (IsaackMwenda) [reply docker-run.sh:13]: Fixed in future pr

**Pull request merged**
